src/main/resources/category/kotlin/common.xml

```diff
@@ -292,5 +291,39 @@ class Foo {
         </example>
     </rule>
 
+    <rule name="AvoidRecreatingDateTimeFormatter"
+          message="Avoid recreating DateTimeFormatter, it is relatively expensive."
+          class="net.sourceforge.pmd.lang.rule.XPathRule" dfa="false" language="kotlin"
+          typeResolution="true" externalInfoUrl="https://github.com/jborgers/PMD-jPinpoint-rules/tree/master/docs/JavaCodePerformance.md#idtf02">
+        <description>
+            Problem: Recreating a DateTimeFormatter is relatively expensive.&#13;
+            Solution: org.joda.time.format.DateTimeFormatter or Java 8 java.time.DateTimeFormatter is thread-safe and can be shared among threads. Create the
+            formatter from a pattern only once, to initialize a constant or instance field.
+            (jpinpoint-rules)</description>
+        <priority>2</priority>
+        <properties>
+            <property name="tag" value="jpinpoint-rule" type="String" description="for-sonar"/>
+            <property name="xpath">
+                <value>
+                    <![CDATA[
+(: check if java.text imports exists as filter for same named classes in other packages :)
+//ImportHeader[.//T-Identifier[@Text='org'] and .//T-Identifier[@Text='joda'] and .//T-Identifier[@Text='time'] and .//T-Identifier[@Text='format']][1]/ancestor::KotlinFile
```

Comment:
is this about java.text (1st line) or joda.time (2nd line)?
Should it not hold for both cases? So match on at least one of them?
Include threeten.bp.format? Maybe not, old.

Reply:
In the test file there is: 

    val ftor: DateTimeFormatter = org.threeten.bp.format.DateTimeFormatter.ofPattern("yyMMdd") // bad

This will match because we do not check the fully qualified name with packages names: `org.threeten.bp.format.`. So any fully specified custom DateTimeFormatter will have a hit.

We might be able to add this to a function that checks imports and fully classified class names.

For now `threeten` is not explicitly included.